facebook/metro
# Support for specifying a custom `require.js` file to be used.

**Summary**
In some cases, we may need to make some changes to the default polyfill file named `require.js`. This commit supports a custom `require.js` file.


**Test plan**
1. Specify a custom `require.js` and add it into the result of config.serializer.getPolyfills().
You should define options to override default config.
``` javascript
let defaultConfigOverrides = {
        resolver: {
          resolverMainFields: ['browser', 'main'],
          blacklistRE: createBlacklist([/.*\/__fixtures__\/.*/]),
        },
        serializer: {
          getModulesRunBeforeMainModule: () => [
            require.resolve('path/of/InitializeCore.js'),
          ],
          getPolyfills: () => [
            require.resolve('path/of/Object.es6.js'),
            ...
            require.resolve('path/of/require.js'),
          ]
        },
        transformer: {
          babelTransformerPath: require.resolve('metro/src/reactNativeTransformer'),
        },
        watchFolders: [],
      };
```
Then use it
``` javascript
const {loadConfig} = require('metro-config');
let config = await loadConfig(argv, defaultConfigOverrides);
```
2. Configure server with useCustomPolyfill whose value is true
3. Build bundle
4. Your code will replace the default code in the bundle.
